### PR TITLE
improve rowSelection.selections of Table

### LINKED ISSUE #5246: Allow Custom Selection Dropdown in Table to be toggled off
#### Environment(required)

- antd version: 2.8.0
- OS and its version: OSX El Capitan
- Browser and its version: Chrome v56

#### What did you do? Please provide steps to re-produce your problem.

I updated my web app from Ant 2.7.4 to 2.8.0.

#### What do you expected?

I expected no visual changes.  Since this new feature was added, it would be nice to have it disabled by default and opt-in.

#### What happen?

New Feature introduced in 2.8.0 - https://github.com/ant-design/ant-design/pull/4962
I noticed an additional dropdown icon that shows up on my Ant Table next to a checkbox.  When expanded, the dropdown shows two items, both in Chinese.

#### Re-producible online demo

https://ant.design/components/table/

### PR DESCRIPTION
**Pull request by infeng**
close #5246.

### CODE REVIEW

File: components/table/SelectionCheckboxAll.tsx [hunk 1]
```diff
@@ -20,7 +21,7 @@ export interface SelectionCheckboxAllProps {
   data: any[];
   prefixCls: string | undefined;
   onSelect: (key: string, index: number, selectFunc: any) => void;
-  selections: SelectionDecorator[];
+  selections: SelectionDecorator[] | boolean | undefined;
```
Comment (benjycui) [inline components/table/SelectionCheckboxAll.tsx]: `selections?` means it could be undefined..

File: components/table/SelectionCheckboxAll.tsx [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { Store } from './createStore';
 import Dropdown from '../dropdown';
 import Menu from '../menu';
 import Icon from '../icon';
+import classnames from 'classnames';
```
Comment (benjycui) [inline components/table/SelectionCheckboxAll.tsx]: Camel case `classNames`.

File: components/table/SelectionCheckboxAll.tsx [hunk 3]
```diff
@@ -151,36 +152,45 @@ export default class SelectionCheckboxAll extends React.Component<SelectionCheck
...
+          </div>
+        </Dropdown>
+      );
+    }
 
     return (
       <div className={selectionPrefixCls}>
         <Checkbox
-          className={`${selectionPrefixCls}-select-all`}
+          className={classNames(`${selectionPrefixCls}-select-all`, {custom: customSelections})}
```
Comment (afc163) [inline components/table/SelectionCheckboxAll.tsx]: 1. custom => ${selectionPrefixCls}-select-custom
2. should have whitespaces inside `{}`

**Review by benjycui — APPROVED** [review]

**Comment by afc163** [comment]
conflicted

**Pull request merged**